### ISSUE TITLE
Add support for GraphQL subscriptions

### ISSUE THREAD

**Issue by grxy** [gramps-graphql/gramps-express#52]
My team would love to use the GrAMPS architecture, but we're blocked by not having subscription support. It would be great if we could have a basic subscription implementation so data sources can add their own subscriptions.

**Comment by jlengstorf** [comment]
@grxy This is on our roadmap, but not highly prioritized — however, if you wanted to add subscriptions to https://github.com/gramps-graphql/gramps (which is on its way to 1.0 right now), PRs are extremely welcome.

If your goal is to just add the basics (e.g. you can define a `subscription mySub` to the schema), it would be a relatively small PR.

**Comment by grxy** [comment]
@jlengstorf Is the https://github.com/gramps-graphql/gramps repo going to replace this one in the long term? Or is it mainly a separation of functionality so that things can be more maintainable?

**Comment by grxy** [comment]
@jlengstorf For now, we've decided to do all of our queries and mutations in GrAMPS while defining our subscriptions separately and calling `mergeSchemas` somewhere in the process. We definitely want to use the new pattern in the long term, but in the short term, it would be nice to have subscription support added in the middleware package. That being said, would you be open to reviewing a PR for this package that provides basic support for subscriptions as a short-term solution and then porting that functionality to the new gramps package as it evolves towards 1.x?

**Comment by jlengstorf** [comment]
@grxy For sure! I think we're almost ready with the 1.0 version of GrAMPS (see [this comment](https://github.com/gramps-graphql/gramps/issues/26#issuecomment-352114348) for details). 

One note: there's a breaking change from 0.x to 1.x in the shape of data sources. Depending on how many data sources you have right now, it may be worth making the refactor sooner rather than later to limit the amount of headache you're up against later.

If the 1.0 pattern works for you, I'd really appreciate PRs against `@gramps/gramps` instead. Data source changes notwithstanding, the GrAMPS core should be a drop-in replacement.

If that doesn't work for you, I'm happy to accept PRs against this repo in the short term.

Thanks!

**Comment by grxy** [comment]
@jlengstorf Awesome! I will take a look at 1.x first to see if that works for our use case.

**Comment by grxy** [comment]
@jlengstorf I don't see mutations added by default in the new package. Are you not planning to support them out of the box or is that functionality just pending.

**Comment by jlengstorf** [comment]
@grxy Anything that's parseable by Apollo _should_ be supported in the new package. Since we're using `mergeSchemas` instead of string concatenation, we don't need to declare anything up front. If you want to give this a shot and write a subscription definition in a data source, I'd love to hear if this works as expected. 😄

**Comment by grxy** [comment]
@jlengstorf I will do that today and let you know.

**Comment by grxy** [comment]
@jlengstorf Our data sources have custom scalar types such that I can't build one of our subscriptions without it. I've created an issue in the other project to address the problem: https://github.com/gramps-graphql/gramps/issues/41

**Comment by jlengstorf** [comment]
@grxy It looks like @ecwyne is operating at Usain Bolt-like speeds today. 😄  Let me know if you're still having trouble after that fix went in.

**Comment by grxy** [comment]
@jlengstorf Will check once that fix is published.

**Comment by grxy** [comment]
@jlengstorf So I was successfully able to get subscriptions to work within a Gramps data source. I was, however only able to set it up such that the resolvers live in the stitching config, not in the main resolvers (mapResolvers was throwing an error). Shall I create an issue in the new repo to address the problem I encountered?

**Comment by jlengstorf** [comment]
@grxy Please do! I'll close this issue, since it sounds like subscriptions are working (ish). 😄

**Comment by grxy** [comment]
@jlengstorf Finally got around to opening an issue: https://github.com/gramps-graphql/gramps/issues/58
